Galaxus.Functional/(Option)/AsyncOptionExtensions.cs

```diff
@@ -77,4 +77,144 @@ public static async Task<T> UnwrapOrElseAsync<T>(this Task<Option<T>> self, Func
 
         return option.Unwrap();
     }
+
+    /// <inheritdoc cref="Option{T}.Match" />
+    public static async Task MatchAsync<T>(this Option<T> self, Func<T, Task> onSome, Func<Task> onNone)
+    {
+        if (self.IsSome)
```

Comment:
Instead of copying the logic from `Option.Match`, we could just call it - right?

```csharp
public static async Task MatchAsync<T>(this Option<T> self, Func<T, Task> onSome, Func<Task> onNone)
{
    await self.Match(onSome, onNone);
}
```